server/src/main/java/de/uftos/repositories/ucdl/parser/DefinitionParser.java

```diff
@@ -0,0 +1,691 @@
+          throw new ParseException("Variable name \"" + varName.value()
+              + "\" does already exist inside this namespace!");
+        }
+        SetDto set = (SetDto) buildAst(root.jjtGetChild(1), parameters);
+        variableDefinition.add(set); //set
+        OperatorDto variable = new OperatorDto(UcdlToken.OF, variableDefinition);
+
+        parameters.put(varName.value(), set.type()); //adding the new variable
+        AbstractSyntaxTreeDto body = buildAst(root.jjtGetChild(1), parameters);
+        parameters.remove(varName.value()); //removing the new variable
+
+        return new QuantifierDto(UcdlToken.EXISTS, variable, body);
+      }
+      case "IMPLIES" -> {
+        AbstractSyntaxTreeDto firstArgument = buildAst(root.jjtGetChild(0), parameters);
+        if (root.jjtGetChild(1).jjtGetNumChildren() > 0) {
+          AbstractSyntaxTreeDto secondArgument =
+              buildAst(root.jjtGetChild(1).jjtGetChild(0), parameters);
+          List<AbstractSyntaxTreeDto> params = new ArrayList<>();
+          params.add(firstArgument);
+          params.add(secondArgument);
+          return new OperatorDto(UcdlToken.IMPLIES, params);
+        }
+        return firstArgument;
+      }
+      case "OR" -> {
+        AbstractSyntaxTreeDto firstArgument = buildAst(root.jjtGetChild(0), parameters);
+        if (root.jjtGetChild(1).jjtGetNumChildren() > 0) {
+          AbstractSyntaxTreeDto secondArgument =
+              buildAst(root.jjtGetChild(1).jjtGetChild(0), parameters);
+          List<AbstractSyntaxTreeDto> params = new ArrayList<>();
+          params.add(firstArgument);
+          params.add(secondArgument);
+          return new OperatorDto(UcdlToken.OR, params);
+        }
+        return firstArgument;
+      }
+      case "AND" -> {
+        AbstractSyntaxTreeDto firstArgument = buildAst(root.jjtGetChild(0), parameters);
+        if (root.jjtGetChild(1).jjtGetNumChildren() > 0) {
+          AbstractSyntaxTreeDto secondArgument =
+              buildAst(root.jjtGetChild(1).jjtGetChild(0), parameters);
+          List<AbstractSyntaxTreeDto> params = new ArrayList<>();
+          params.add(firstArgument);
+          params.add(secondArgument);
+          return new OperatorDto(UcdlToken.AND, params);
+        }
+        return firstArgument;
+      }
+      case "NOT" -> {
+        if (((SimpleNode) root).jjtGetFirstToken().image.equals("NOT")) {
+          List<AbstractSyntaxTreeDto> param = new ArrayList<>();
+          param.add(buildAst(root.jjtGetChild(0), parameters));
+          return new OperatorDto(UcdlToken.NOT, param);
+        }
+        return buildAst(root.jjtGetChild(0), parameters);
+      }
+      case "ISEMPTY" -> {
+        List<AbstractSyntaxTreeDto> param = new ArrayList<>();
+        param.add(buildAst(root.jjtGetChild(0), parameters));
+        return new OperatorDto(UcdlToken.IS_EMPTY, param);
+      }
+      case "ELEMENTINSETOREQUATION" -> {
```

Comment:
Is it allowed to have underscores in tokens? That would help with readability *a lot*, especially for `ELEMENT_IN_SET_OR_EQUATION`

Reply:
Yes, '_' is allowed. This will require generating a new parser.